agent/doc/UML/agent_device_management_design.puml

```diff
@@ -154,4 +154,45 @@ LinuxScanner -u-|> EnvCapabilityScanner
 MacOSScanner -u-|> EnvCapabilityScanner
 
 EnvCapabilityScanner .. EnvCapability: produce
+@enduml
+
+@startuml agent_capability_design
+
+class AgentManagementService {
+    EnvInfo envInfo;
+    List<ServiceCapability> serviceCapability;
+    detectAndRegisterServiceCapability(ServiceCapability)
+}
+AgentManagementService -r-> EnvCapabilityDiscoveryService
+class TestRunnerConfig
+class TestRunner{
+    ServiceCapability serviceCapability;
+}
+class DeviceDriverConfig
+class DeviceDriver{
+    ServiceCapability serviceCapability;
+}
+TestRunnerConfig ..> TestRunner : create
+TestRunnerConfig -r-> AgentManagementService
+DeviceDriverConfig ..> DeviceDriver : create
+DeviceDriverConfig -r-> AgentManagementService
+TestRunner --> ServiceCapability
+DeviceDriver --> ServiceCapability
+
+class ServiceCapability{
```

Comment:
Why don't we use the EnvCapbility?

Reply:
Maybe we don't need to create a new class as service capability, just leverage the https://github.com/microsoft/HydraLab/blob/main/agent/src/main/java/com/microsoft/hydralab/agent/environment/EnvCapability.java. And make it a requirement, provide a method called meet(EnvCapability) in EnvCapability to see if the requirement meet the provided capability. @zhou9584